### PR TITLE
KEYCLOAK-18503 Regex Policy for authorization service

### PR DESCRIPTION
**Pull request by y-tabata**
https://issues.redhat.com/browse/KEYCLOAK-18503

### CODE REVIEW

**Comment by y-tabata** [comment]
Docs PR: https://github.com/keycloak/keycloak-documentation/pull/1212

File: authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/regex/RegexPolicyProvider.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public void evaluate(Evaluation evaluation) {
+        AuthorizationProvider authorizationProvider = evaluation.getAuthorizationProvider();
+        RegexPolicyRepresentation policy = representationFunction.apply(evaluation.getPolicy(), authorizationProvider);
+        Attributes.Entry targetClaim = evaluation.getContext().getIdentity().getAttributes().getValue(policy.getTargetClaim());
+
+        if (targetClaim == null) {
+            return;
+        }
+
+        Pattern pattern = Pattern.compile(policy.getPattern());
```
Comment (pedroigor) [inline authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/regex/RegexPolicyProvider.java:56]: Ideally, we could compile the `Pattern` only once. But I'm OK as is and improve it later.

File: authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/regex/RegexPolicyProvider.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Override
+    public void close() {
+    }
+
+    @Override
+    public void evaluate(Evaluation evaluation) {
+        AuthorizationProvider authorizationProvider = evaluation.getAuthorizationProvider();
+        RegexPolicyRepresentation policy = representationFunction.apply(evaluation.getPolicy(), authorizationProvider);
+        Attributes.Entry targetClaim = evaluation.getContext().getIdentity().getAttributes().getValue(policy.getTargetClaim());
```
Comment (pedroigor) [inline authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/regex/RegexPolicyProvider.java:50]: In the future, we could potentially extend it to also consider contextual attributes (those obtained from the `EvaluationContext`).

**Review by pedroigor — APPROVED** [review]
@y-tabata Nice one. Tks.

**Comment by y-tabata** [comment]
@pedroigor Thanks!

**Pull request merged**